lib/migration.js

```diff
@@ -461,8 +461,8 @@ function mixinMigration(MySQL, mysql) {
 
   function columnType(p, defaultType) {
     var dt = defaultType;
-    if (p.dataType) {
-      dt = String(p.dataType);
```

Comment:
@pluswave This affects the backwards compatibility here. Breaking a bunch of tests related to it. I agree with your changes but it should go with the purpose of the fix. 

Let's say we have the model definition:
```
"modelName": {
"type" : "someType",
"mysql" : {
"dataType": "someOtherType"
}
}
```
**When doing auto migration / auto update, it should use someOtherType instead of
the default map of someType.**

With your patch, it will try to always look for `mysql.OtherType` and if it does not exists it would just `return dt;`, not acceptable because the type could have been also set by `"type": "someType"`.

Hence, I would add some tweaks to accept both ways, something like this:
```
    if (p.mysql && p.mysql.dataType) {
      dt = String(p.mysql.dataType);
    }
    if (p.dataType) {
      dt = String(p.dataType);
    }
```